src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/ArcParser.java

```diff
@@ -32,19 +32,173 @@
    *MicrosoftOfficeWebServer: 5.0_Pub
    *X-Powered-By: ASP.NET
    */
-  public static ArcEntry getArcEntry(String arcFilePath, long arcEntryPosition) throws Exception {
+  public static ArcEntry getArcEntry(String arcFilePath, long arcEntryPosition, boolean loadBinary) throws Exception {
     RandomAccessFile raf= null ;
     try{
 
       if (arcFilePath.endsWith(".gz")){ //It is zipped
-        return getArcEntryZipped(arcFilePath, arcEntryPosition);                       
+        return getArcEntryZipped(arcFilePath, arcEntryPosition, loadBinary);                       
       }
+      else {
+          return getArcEntryNotZipped(arcFilePath, arcEntryPosition, loadBinary);
+      }      
+    }
+    catch(Exception e){
+      throw e;
+    }
+    finally{
+      if (raf!= null){
+        raf.close();
+      }
+    }
+  }
+
+  
+  public static ArcEntry getArcEntryNotZipped(String arcFilePath, long arcEntryPosition, boolean loadBinary) throws Exception {
+      RandomAccessFile raf= null ;
       ArcEntry arcEntry = new ArcEntry();
-      StringBuffer headerLinesBuffer = new StringBuffer();
+      arcEntry.setFormat(ArcEntry.FORMAT.ARC);
+      arcEntry.setSourceFilePath(arcFilePath);
+      arcEntry.setOffset(arcEntryPosition);
+      
+      try{        
+      
+        raf = new RandomAccessFile(new File(arcFilePath), "r");
```

Comment:
Consider using the Java 1.7 `Closeable` feature and write
```
    try (RandomAccessFile raf = new RandomAccessFile(new File(arcFilePath), "r")){
```
instead of checking if `raf`has been created in the `finally` block.

Reply:
Fixed in both ArcParser and WarcParser